src/resources/uart_esp32.cc

```diff
@@ -428,12 +435,32 @@ UART_ISR_INLINE void UartResource::disable_rx_interrupts() {
   disable_interrupt_index(UART_TOIT_INTR_RX_TIMEOUT);
 }
 
-UART_ISR_INLINE void UartResource::enable_tx_empty_interrupts() {
+UART_ISR_INLINE void UartResource::enable_tx_interrupts(bool begin) {
```

Comment:
I think the name is now a bit confusing: it's true that the TXFIFO interrupt is enabled, but the TX_DONE is disabled.
Maybe: `set_tx_start_interrupts`
and then `set_tx_stop_interrupts`

Reply:
Reworked.